h2o-r/h2o-package/R/models.R

```diff
@@ -2376,26 +2413,69 @@ h2o.coef_norm <- function(object, predictorSize=-1) {
   if (is(object, "H2OModel") &&
       (object@algorithm %in% c("glm", "gam", "coxph", "modelselection"))) {
     if (object@algorithm == "modelselection") {
-      modelIDs <- object@model$best_model_ids
-      numModels = length(modelIDs)
-      mode <- object@parameters$mode
-      maxPredNumbers <- length(object@model$best_model_predictors[[numModels]])
-      if (predictorSize == 0)
-        stop("predictorSize (predictor subset size) must be between 0 and the total number of predictors used.")
-      if (predictorSize > maxPredNumbers)
-        stop("predictorSize (predictor subset size) cannot exceed the total number of predictors used.")
-      if (predictorSize > 0) { # subset size was specified 
+      if (object@allparameters$mode == "maxrsweep" &&
+          !object@allparameters$build_glm_model) {
+        numModels <- length(object@model$best_r2_values)
+        maxPredNumbers <- object@parameters$max_predictor_number
+        if (predictorSize == 0)
+          stop(
+            "predictorSize (predictor subset size) must be between 0 and the total number of predictors used."
+          )
+        if (predictorSize > maxPredNumbers)
+          stop(
+            "predictorSize (predictor subset size) cannot exceed the total number of predictors used."
+          )
+        if (predictorSize < 0) {
+          coeffs <- vector("list", numModels)
+          for (index in seq(numModels)) {
+            coeffs[[index]] <-
+              structure(
+                object@model$coefficient_values_normalized[[index]],
+                names = object@model$coefficient_names[[index]]
+              )
+          }
+          return(coeffs)
+        } else {
+          return(structure(object@model$coefficient_values_normalized[[predictorSize]],
+                    names = object@model$coefficient_names[[predictorSize]]))
+        }
+        
+        if (predictorSize < 0) {
+          structure(names=object@model$coefficient_names, object@model$coefficient_values_normalized)
+        } else {
+          structure(names=object@model$coefficient_names[[predictorSize]], object@model$coefficient_values_normalized[[predictorSize]])    
+        }        
+      } else {
+        modelIDs <- object@model$best_model_ids
+        numModels = length(modelIDs)
+        mode <- object@parameters$mode
+        maxPredNumbers <- numModels
+        if (predictorSize == 0)
+          stop(
+            "predictorSize (predictor subset size) must be between 0 and the total number of predictors used."
+          )
+        if (predictorSize > maxPredNumbers)
+          stop(
+            "predictorSize (predictor subset size) cannot exceed the total number of predictors used."
+          )
```

Comment:
I'm wondering if the logic around here could be refactored since there appears to be quite a lot of duplication.
```suggestion
        stopifnot(
            "predictorSize (predictor subset size) must be between 0 and the total number of predictors used." = predictorSize != 0,
            "predictorSize (predictor subset size) cannot exceed the total number of predictors used." = predictorSize <= maxPredNumbers
        )
```